src/nl/hannahsten/texifyidea/editor/folding/LatexCommentFoldingBuilder.kt

```diff
@@ -0,0 +1,109 @@
+import nl.hannahsten.texifyidea.util.childrenOfType
+
+/**
+ * Enables folding of multiple comments on successive lines.
+ *
+ * @author jojo2357
+ */
+class LatexCommentFoldingBuilder : FoldingBuilderEx(), DumbAware {
+
+    override fun isCollapsedByDefault(node: ASTNode) = false
+
+    override fun getPlaceholderText(node: ASTNode): String {
+        val parsedText = node.text.trim()
+        return if (parsedText.length > 9) parsedText.substring(0, 8) + "..."
+        else parsedText
+    }
+
+    override fun buildFoldRegions(root: PsiElement, document: Document, quick: Boolean): Array<FoldingDescriptor> {
+        val descriptors = ArrayList<FoldingDescriptor>()
+        val comments = root.childrenOfType(PsiComment::class)
+        val whitespaces = root.childrenOfType(PsiWhiteSpace::class)
+
+        val whitespaceLocations = ArrayList(whitespaces.map { TextRange(it.startOffset, it.endOffset) })
+
+        // meld friends
+        for (i in whitespaceLocations.size - 1 downTo 1) {
+            if (whitespaceLocations[i].startOffset == whitespaceLocations[i - 1].endOffset) {
+                whitespaceLocations[i - 1] =
+                    TextRange(whitespaceLocations[i - 1].startOffset, whitespaceLocations[i].endOffset)
+                whitespaceLocations.removeAt(i)
+            }
+        }
+
+        var collectedTextRange: TextRange? = null
+        var parentCollapse: PsiElement? = null
+
+        for (comment in comments) {
+
+            // Initialization: start with the first comment in a possible sequence
+            if (collectedTextRange == null) {
+                collectedTextRange = TextRange(comment.startOffset, comment.endOffset)
+                parentCollapse = comment.originalElement
+            }
+            else {
+                // If the next comment follows directly after the previous one, add it to the sequence
+                if (whitespaceLocations.any { it.startOffset == collectedTextRange!!.endOffset && it.endOffset == comment.startOffset }) {
+                    collectedTextRange = TextRange(collectedTextRange.startOffset, comment.endOffset)
+                }
+                else {
+                    // Otherwise, stop the sequence and fold what we have so far
+                    parentCollapse?.let {
+                        descriptors.add(
+                            FoldingDescriptor(
+                                parentCollapse!!,
+                                TextRange(collectedTextRange!!.startOffset, collectedTextRange!!.endOffset)
+                            )
+                        )
+                    }
+                    collectedTextRange = TextRange(comment.startOffset, comment.endOffset)
+                    parentCollapse = comment.originalElement
+                }
+            }
+            // todo why is this code here?
+            if (!whitespaceLocations.any { it.startOffset == collectedTextRange!!.endOffset }) {
```

Comment:
It is not clear to me why this piece of code is here, could you add a comment and a test to clarify this?

Reply:
Resolve if rectified

Reply:
Still not clear, because if there is no whitespace then either there's going to be comment after that (handled in the block above) or not (handled in the block at the end). I'll remove this part